rescript/ncbi_datasets.py

```diff
@@ -0,0 +1,180 @@
+        with open(str(genomes), 'a') as fin:
+            for f in genome_seq_fps:
+                seq = skbio.read(
+                    f, format='fasta', constructor=skbio.DNA, lowercase=True
+                )
+                skbio.io.write(seq, format='fasta', into=fin)
+
+        # find and move all the gff files
+        loci = LociDirectoryFormat()
+        loci_fps = glob.glob(
+            os.path.join(tmp, 'ncbi_dataset', 'data', '*', 'genomic.gff'))
+        for f in loci_fps:
+            _id = f.split('/')[-2]
+            shutil.move(f, os.path.join(str(loci), f'{_id}_loci.gff'))
+
+        # find and move all the protein translations
+        proteins = ProteinsDirectoryFormat()
+        protein_fps = glob.glob(
+            os.path.join(tmp, 'ncbi_dataset', 'data', '*', 'protein.faa'))
+        for f in protein_fps:
+            _id = f.split('/')[-2]
+            shutil.move(f, os.path.join(
+                str(proteins), f'{_id}_proteins.fasta'))
+    return genomes, loci, proteins
+
+
+def _fetch_taxonomy(all_acc_ids, all_tax_ids):
+    manager = Manager()
+    taxa, bad_accs = get_taxonomies(
+        taxids={k: v for k, v in zip(all_acc_ids, all_tax_ids)},
+        ranks=_default_ranks, rank_propagation=True,
+        logging_level='INFO', n_jobs=2, request_lock=manager.Lock()
+    )
+    # technically, this should never happen as the taxa accession
+    # numbers are extracted from the API response
+    if bad_accs:
+        raise Exception('Invalid taxonomy accession numbers were found: '
+                        f'{", ".join(bad_accs)}. Please check your query '
+                        f'and try again.')
+    taxa = pd.DataFrame(taxa, index=['Taxon']).T
+    taxa.index.name = 'Feature ID'
+    return taxa
+
+
+def get_ncbi_genomes(
+        taxon: str,
+        assembly_source: str = 'refseq',
+        assembly_levels: List[str] = ['complete_genome'],
+        only_reference: bool = True,
+        tax_exact_match: bool = False,
+        page_size: int = 20,
+) -> (DNAFASTAFormat, LociDirectoryFormat,
+      ProteinsDirectoryFormat, pd.DataFrame):
+    # we use a deepcopy of assembly_levels because the new versions of
+    # the NCBI datasets toolkit somehow, magically, update it what
+    # causes a conflict in the q2cli resulting in a failure to dump
+    # the action's params to provenance
+    assembly_descriptors_params = {
+        'assembly_levels': deepcopy(assembly_levels),
+        'assembly_source': assembly_source,
+        'only_reference': only_reference, 'page_size': page_size,
+        'taxon': taxon, 'tax_exact_match': tax_exact_match
+    }
+    with DatasetsApiClient() as api_client:
+        api_instance = nd.GenomeApi(api_client)
+
+        all_acc_ids, all_tax_ids = _get_assembly_descriptors(
+            api_instance=api_instance, **assembly_descriptors_params
+        )
+
+        api_response = api_instance.download_assembly_package(
+            all_acc_ids,
+            # chromosomes=['MT'],
```

Comment:
Actually, I left it there on purpose to remember that in the future this can be generalized to more genomes with multiple chromosomes where you can pick which chromosome you want to download. I can remove it though and make a note elsewhere if you prefer ;)

Reply:
Good point actually, thanks - I deleted it now and will just open an issue after this is merged to keep track there.